dom.bs

```diff
@@ -3635,6 +3666,67 @@ method steps are:
 <ol>
  <li><p>Assert: either <var>addedNodes</var> or <var>removedNodes</var> <a for=set>is not empty</a>.
 
+ <li>
+  <p>If either <var>options</var>'s' {{MutationObserverInit/elementFilter}} or
+  {{MutationObserverInit/elementByAttributeFilter}} is present, then:
+
+  <ol>
+   <li><p>Let <var>filteredAddedNodes</var> be an empty <a for=/>list</a>.
+
+   <li><p>Let <var>filterRemovedNodes</var> be an empty <a for=/>list</a>.
+
+   <li>
+    <p>For each <var>node</var> in <var>addedNodes</var></a>:
+
+    <ol>
+     <li><p>Let <var>nodeAttributesList</var> be the set of <a for=Attr>local name</a>s of <var>node</var>'s
+     <a>attribute</a>s.
+
+     <li>
+      <p>If none of the following are true
+
+      <ul class=brief>
+       <li><var>options</var>'s {{MutationObserverInit/elementFilter}} is present, and the
+       <a>node</a>'s <var>localName</var> is in {{MutationObserverInit/elementFilter}}
+
+       <li><var>options</var>'s {{MutationObserverInit/elementByAttributeFilter}} is present, and
+       <var>options</var>'s {{MutationObserverInit/elementByAttributeFilter}} contains any of the
+       attribute name in <var>nodeAttributesList</var>
+      </ul>
+
+      <p>then <a for=list>append</a> <var>node</var> to <var>filteredAddedNodes</var>.
+    </ol>
+
+   <li>
+    <p>For each <var>node</var> in <var>removedNodes</var></a>:
+
+    <ol>
+     <li><p>Let <var>nodeAttributesList</var> be the set of <a for=Attr>local name</a>s of <var>node</var>'s
+     <a>attribute</a>s.
+
+     <li>
+      <p>If none of the following are true
+
+      <ul class=brief>
+       <li><var>options</var>'s {{MutationObserverInit/elementFilter}} is present, and the
+       <a>node</a>'s <var>localName</var> is in {{MutationObserverInit/elementFilter}}
+
+       <li><var>options</var>'s {{MutationObserverInit/elementByAttributeFilter}} is present, and
+       <var>options</var>'s {{MutationObserverInit/elementByAttributeFilter}} contains any of the
+       attribute names in <var>nodeAttributesList</var>
+      </ul>
+
+      <p>then <a for=list>append</a> <var>node</var> to <var>filterRemovedNodes</var>.
```

Comment:
I must be confused here.
Don't we want something like
if elementFilter is present, and elementByAttributeFilter is not present and localName is in elementfilter, or if 
elementByAttributeFilter is present and elementFilter is not present and elementByAttributeFilter contains any of the attribute names in nodeAttribute list ...  or combination of those two